File: templates/ruby/lib/http_client/response_handler.rb
module {{call .Fnc.camelize .Pkg.Name}}

  module HttpClient

    # ResponseHandler takes care of decoding the response body into suitable type
    class ResponseHandler

      def self.get_body(response)
        type = response.headers["content-type"]
        body = response.body
{{if .Api.Response.Formats.Json}}
        # Response body is in JSON
        if type and type.include?("json")
          begin
            body = JSON.parse body
          rescue JSON::ParserError
            return body
          end
        end
{{end}}
        return body
      end

    end

  end

end

File: templates/python/lib/http_client/response_handler.py
class ResponseHandler(object):

    """ResponseHandler takes care of decoding the response body into suitable type"""

    @staticmethod
    def get_body(response):
        typ = response.headers.get('content-type')
        body = response.text
{{if .Api.Response.Formats.Json}}
        # Response body is in JSON
        if typ.find('json') != -1:
            body = response.json()
{{end}}
        return body
